tests/functional/c/consider/consider_refactoring_into_while_condition.py

```diff
@@ -0,0 +1,78 @@
+"""Emit a message for breaking out of a while True loop immediately."""
+# pylint: disable=missing-function-docstring,missing-class-docstring,unrecognized-inline-option,invalid-name
+
+class Issue8015:
+    def bad(self):
+        k = 1
+        while True:  # [consider-refactoring-into-while-condition]
+            if k == 10:
+                break
+            k += 1
+
+    def another_bad(self):
+        current_scope = None
+        while True:  # [consider-refactoring-into-while-condition]
+            if current_scope is None:
+                break
+            current_scope = True
+
+    def good(self):
+        k = 1
+        while True:
+            k += 1
+            if k == 10:
+                break
+
+    def test_error_message(self):
+        a_list = [1,2,3,4,5]
+        # Should recommend `while a_list`
+        while True:   # [consider-refactoring-into-while-condition]
+            if not a_list:
+                break
+            a_list.pop()
+
+    def test_error_message_2(self):
+        a_list = []
+        # Should recommend `while not a_list`
+        while True:   # [consider-refactoring-into-while-condition]
+            if a_list:
+                break
+            a_list.append(1)
+
+    def test_error_message_3(self):
+        a_list = []
+        # Should recommend `while not a_list`
+        while True:   # [consider-refactoring-into-while-condition]
+            if a_list is not None:

```

Comment:
`not a_list` and `not a_list is None` are not equivalent. In this case, the former will fail immediately and the latter will never fail. I think it would be safer (and easier) to simply negate the condition, even if that would lead to something like a double negative.

Reply:
If the suggested change would be unidiomatic, that could be caught with a separate checker on a subsequent Pylint pass. Maybe requiring multiple passes could be considered undesirable. I don't know the best way to handle that.

Reply:
yeah i'm stumped on this dilemma as well. ideally I want to generate an idiomatic suggestion which would be more useful

Reply:
@nickdrozd ,

wouldn't 

` if a_list is not None` be compacted to 

```py
while not (a_list is not None)
```

since `a_list is not None` can be expressed as simply `a_list` i would assume this recommendation of `while not a_list` to be correct

Reply:
The empty list is falsy, but it still isn't `None`. Try this:

```python
a_list = []

while not (a_list is not None):  # fails immediately
    print('asdf')  # unreachable

while not a_list:  # never fails, infinite loop
    print('zxcv')
```

Reply:
Ah okay, I see where you're coming from.

In this case, however I wanted to provide an example where sometimes a variable/attribute is checked to be None and if so breaks out of the loop.

In this case we can refactor it to while (var/attr)

I'll update the test cases to make it more clear. thanks